p2p/kademlia/dao/sqlite/queries_test.go

```diff
@@ -0,0 +1,209 @@
+	"github.com/stretchr/testify/assert"
+)
+
+var ctx = context.Background()
+
+func setupDB() *sql.DB {
+	db, _ := sql.Open("sqlite3", ":memory:")
+	Migrate(ctx, db)
+	return db
+}
+
+func TestMigrate(t *testing.T) {
+	db, _ := sql.Open("sqlite3", ":memory:")
+	type args struct {
+		db *sql.DB
+	}
+	tests := []struct {
+		name      string
+		args      args
+		assertion assert.ErrorAssertionFunc
+	}{
+		{
+			name:      "running migrations",
+			args:      args{db},
+			assertion: assert.NoError,
+		},
+		{
+			name:      "running migrations twice",
+			args:      args{db},
+			assertion: assert.NoError,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.assertion(t, Migrate(ctx, tt.args.db))
+		})
+	}
+}
+
+func TestStore(t *testing.T) {
+	db := setupDB()
+
+	type args struct {
+		db          *sql.DB
+		data        []byte
+		replication time.Time
+		expiration  time.Time
+	}
+	tests := []struct {
+		name      string
+		args      args
+		want      int64
+		assertion assert.ErrorAssertionFunc
+	}{
+		{
+			name:      "storing one key",
+			args:      args{db, []byte("test data"), time.Now(), time.Now()},
+			want:      int64(1),
+			assertion: assert.NoError,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := Store(ctx, tt.args.db, tt.args.data, tt.args.replication, tt.args.expiration)
+			tt.assertion(t, err)
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func TestRetrieve(t *testing.T) {
+	db := setupDB()
+	_, err := Store(ctx, db, []byte("test data"), time.Now(), time.Now())
+	assert.NoError(t, err)
+
+	type args struct {
+		db  *sql.DB
+		key []byte
+	}
+	tests := []struct {
+		name      string
+		args      args
+		want      []byte
+		assertion assert.ErrorAssertionFunc
+	}{
+		{
+			name:      "good key",
+			args:      args{db, crypto.GetKey([]byte("test data"))},
+			want:      []byte("test data"),
+			assertion: assert.NoError,
+		},
+		{
+			name:      "bad key",
+			args:      args{db, []byte("12312312313123")},
+			want:      []byte(nil),
+			assertion: assert.NoError,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := Retrieve(ctx, tt.args.db, tt.args.key)
+			tt.assertion(t, err)
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func TestExpireKeys(t *testing.T) {
+	db := setupDB()
+	_, err := Store(ctx, db, []byte("test data 1"), time.Now(), time.Now())
+	assert.NoError(t, err)
+
+	_, err = Store(ctx, db, []byte("test data 2"), time.Now(), time.Now())
+	assert.NoError(t, err)
+
+	type args struct {
+		db *sql.DB
+	}
+	tests := []struct {
+		name      string
+		args      args
+		assertion assert.ErrorAssertionFunc
+	}{
+		{
+			name:      "expire test data",
+			args:      args{db},
+			assertion: assert.NoError,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.assertion(t, ExpireKeys(ctx, tt.args.db))
+		})
+	}
+}
+
+func TestGetAllKeysForReplication(t *testing.T) {
```

Comment:
This test doesn't pass in my environment 
![image](https://user-images.githubusercontent.com/62853952/119736776-ed3c9e00-be86-11eb-8dfa-0bcf9cdaa1de.png)

it passes, when I remove sql condition  `WHERE replication > TIME('now')` from here
```go
	rows, err := db.QueryContext(ctx, "SELECT key FROM keys WHERE replication > TIME('now')")
```
Is it possible that the problem is in the time zone?

Reply:
after an hour tried again and now the test passes.
It is not stable, and then 100% will pop up again, at the worst possible moment.

I think the issue in `TIME('now')` a bit different from `time.Now()`

Reply:
i've removed sql constants and used time.Now() and changed the condition from `> ? ` to `<= ?`.